sim/common/xlen.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
```

Comment:
I don't think you need this file.
Word is XLEN
DWord is XLEN*2
QWord is XLEN*4
if you don't need an X-specific type, just use the actual true types: uint32_t, uint64_t, etc...